### PR TITLE
renalSctructures CECT bundle created

### PR DESCRIPTION
**Pull request by blacky-i**
Fixes # 459

### Description
Adding renalSctructures CECT segmenattion bundle from our latest paper.

### Status
**Ready**

### Please ensure all the checkboxes:

- [x] Codeformat tests passed locally by running `./runtests.sh --codeformat`.
- [ ] In-line docstrings updated.
- [x] Update `version` and `changelog` in `metadata.json` if changing an existing bundle.
- [x] Please ensure the naming rules in config files meet our requirements (please refer to: `CONTRIBUTING.md`).
- [x] Ensure versions of packages such as `monai`, `pytorch` and `numpy` are correct in `metadata.json`.
- [x] Descriptions should be consistent with the content, such as `eval_metrics` of the provided weights and TorchScript modules.
- [x] Files larger than 25MB are excluded and replaced by providing download links in `large_file.yml`.
- [x] Avoid using path that contains personal information within config files (such as use `/home/your_name/` for `"bundle_root"`).

### CODE REVIEW

**Comment by blacky-i** [comment]
Whoops, codeformat is not passed, fixing it

**Comment by blacky-i** [comment]
Now codeformat is passed. 
I didn't notice them, because they are in optional tests. And of course I skipped them) Maybe it is better to move them to necessary tests?

**Comment by wyli** [comment]
/black
thanks for the PR @blacky-i, it's also possible to trigger the auto formatting tool by starting a comment with `/black`. (documentation of contributing should be updated to reflect this, let me create a feature request.)

**Comment by blacky-i** [comment]
Noted about comment. 

Found minor issue though. Model weights published to Yandex.Disk, but to download them it is required to make pull request for `monai/apps/utils.py` file. I will update link to Google Drive temporarily.

**Comment by blacky-i** [comment]
Fixed link

File: models/renalStructures_CECT_segmentation/docs/README.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+### **Tags**
+Segmentation, CT, CECT, Kidney, Renal, Supervised
+
+## **Model Description**
+The model is the SegResNet architecture[1] for volumetric (3D) renal structures segmentation. Input is artery, vein, excretory phases after mutual registration and concatenated to 3 channel 3D tensor.
+
+
+## **Data**
+DICOM data from 41 patients with kidney neoplasms were used. Data included all phases of contrast-enhanced multispiral computed tomography. We split the data: 32 observations for the training set and 9 – for the validation set. At the labeling stage, the arterial, venous, and excretory phases were taken, affine registration was performed to jointly match the location of the kidneys, and noise was removed using a median filter and a non-local means filter. Validation set ip published to Yandex.Disk. You can download via [link](https://disk.yandex.ru/d/pWEKt6D3qi3-aw) or use following command:
```
Comment (yiheng-wang-nv) [inline models/renalStructures_CECT_segmentation/docs/README.md]: Hi @blacky-i , thanks for the contribution. As for the dataset this bundle used, could you please also mention the related license?
see:
https://github.com/Project-MONAI/model-zoo/blob/b245833e22544126f23925eac89da471562467a1/CONTRIBUTING.md#license-format

Here is an example:
https://github.com/Project-MONAI/model-zoo/blob/dev/models/brats_mri_axial_slices_generative_diffusion/docs/data_license.txt

Comment (blacky-i) [reply models/renalStructures_CECT_segmentation/docs/README.md]: Hello, added license information to `docs/` folder

File: models/renalStructures_CECT_segmentation/LICENSE [hunk 2]
```diff
@@ -0,0 +1,201 @@
+ Apache License
```
Comment (yiheng-wang-nv) [inline models/renalStructures_CECT_segmentation/LICENSE]: This line may need to be adjusted as shown in:
https://www.apache.org/licenses/LICENSE-2.0
```
                                 Apache License
                           Version 2.0, January 2004
                        http://www.apache.org/licenses/
```

File: models/renalStructures_CECT_segmentation/docs/README.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+}
+```
+
+## **References**
+
+[1] Myronenko, A. (2019). 3D MRI Brain Tumor Segmentation Using Autoencoder Regularization. In: Crimi, A., Bakas, S., Kuijf, H., Keyvan, F., Reyes, M., van Walsum, T. (eds) Brainlesion: Glioma, Multiple Sclerosis, Stroke and Traumatic Brain Injuries. BrainLes 2018. Lecture Notes in Computer Science(), vol 11384. Springer, Cham. https://doi.org/10.1007/978-3-030-11726-9_28
+
+[2] Chernenkiy, I. М., et al. "Segmentation of renal structures based on contrast computed tomography scans using a convolutional neural network." Sechenov Medical Journal 14.1 (2023): 39-49.https://doi.org/10.47093/2218-7332.2023.14.1.39-49
+
+#### **Tests used for bundle checking**
```
Comment (yiheng-wang-nv) [inline models/renalStructures_CECT_segmentation/docs/README.md:88]: Thanks for adding this section. I will run these checks one by one locally.
In addition, we added unit tests for a part of bundles recently to ensure config files are correct. For example:
https://github.com/Project-MONAI/model-zoo/blob/6122ec9475c87726e14ea0e7a06ffdc3532a1ea2/ci/unit_tests/test_brats_mri_axial_slices_generative_diffusion.py
Are you interested in preparing the tests on this bundle? Feel free to ask if you meeting any issues.
TH
Please note that this is not a hard requirement of contributing a bundle, and we can also merge this PR first and then add unit tests in a separate PR in the future.

Comment (blacky-i) [reply models/renalStructures_CECT_segmentation/docs/README.md:88]: Hello! Sure, I can add bundle test, but I can do it on next week

**Comment by blacky-i** [comment]
@yiheng-wang-nv please, check my PR

File: models/renalStructures_CECT_segmentation/docs/README.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+python -m monai.bundle run training --dont_finetune false --meta_file configs/metadata.json --config_file configs/train.json
+```
+Expected result: finished, Training process started, model variables are restored
+
+#### Execute validation:
+
+##### Download validation data:
+```bash
+cd models/renalStructures_CECT_segmantation
+python -m monai.bundle run download_data --meta_file configs/metadata.json --config_file "['configs/train.json', 'configs/evaluate.json']"
```
Comment (yiheng-wang-nv) [inline models/renalStructures_CECT_segmentation/docs/README.md]: duplicate with https://github.com/Project-MONAI/model-zoo/pull/460/files#diff-ce8c39aacac74a1a44c6b2b8c34d4f843f8f7995f9e8df0c8f20855af1d64e1dR17
may not need this subsection as it has been mentioned in **Data**, and users may run the code twice (the download program does not support skip downloading if the file is existing).

File: models/renalStructures_CECT_segmentation/docs/README.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+```
+Expected result: finished, `data/` folder is created and filled with images.
+
+#####  Run validation script:
+```bash
+python -m monai.bundle run evaluate --meta_file configs/metadata.json --config_file "['configs/train.json', 'configs/evaluate.json']"
+```
+Expected result: finished, `Key metric: val_mean_dice best value: 0.7844631671905518 at epoch: 1` is printed.
```
Comment (yiheng-wang-nv) [inline models/renalStructures_CECT_segmentation/docs/README.md]: Different machines may produce different results. On my trial, I got `0.7844625115394592`. Could you add a few more comments here to tell users that run validation (with provided weights) can get ~0.78446 `val_mean_dice` score?

File: models/renalStructures_CECT_segmentation/docs/README.md [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+
+## **System Configuration**
+The model was trained for 10000 epochs on 2 RTX2080Ti GPUs with [SmartCacheDataset](https://docs.monai.io/en/stable/data.html#smartcachedataset). This takes 1 days and 2 hours, with 4 images per GPU.
```
Comment (yiheng-wang-nv) [inline models/renalStructures_CECT_segmentation/docs/README.md:60]: Could you please also add a line to show what is the minimal GPU requirements?
Reference:
https://github.com/Project-MONAI/model-zoo/tree/b245833e22544126f23925eac89da471562467a1/models/renalStructures_UNEST_segmentation#training-configuration

File: models/renalStructures_CECT_segmentation/docs/README.md [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+python -m monai.bundle run download_data --meta_file configs/metadata.json --config_file "['configs/train.json', 'configs/evaluate.json']"
+```
+Expected result: finished, `data/` folder is created and filled with images.
+
+
+Checking evaluation script
+```bash
+python -m monai.bundle run evaluate --meta_file configs/metadata.json --config_file "['configs/train.json', 'configs/evaluate.json']"
+```
+Expected result: finished, `Key metric: val_mean_dice best value: 0.7844631671905518 at epoch: 1` is printed.
```
Comment (yiheng-wang-nv) [inline models/renalStructures_CECT_segmentation/docs/README.md]: same here, may only need an approximate number

**Review by yiheng-wang-nv — APPROVED** [review]
Thanks @blacky-i for the contribution. This bundle looks good, and all configs are checked locally on my side. Could you please take a look at the above comments? After resolving those minor issues, I think this PR can merge.

File: models/renalStructures_CECT_segmentation/configs/metadata.json [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+    },
+    "monai_version": "1.2.0",
+    "pytorch_version": "1.13.1",
+    "numpy_version": "1.24.3",
+    "optional_packages_version": {
+        "nibabel": "5.1.0",
+        "pytorch-ignite": "0.4.11",
+        "einops": "0.6.1",
+        "fire": "0.5.0",
+        "torchvision": "0.15.2"
```
Comment (yiheng-wang-nv) [inline models/renalStructures_CECT_segmentation/configs/metadata.json]: The CI tests failed because of lib conflicts. Could you double confirm the package versions?
I checked the torchvision and `0.15.2` requires torch 2.0
https://github.com/pytorch/vision#installation

Comment (blacky-i) [reply models/renalStructures_CECT_segmentation/configs/metadata.json]: Yes, you are right. I will change torchvision to match pytorch (I think I had torch 2.0 installed and downgraded it, but without downgrading torchvision).

File: models/renalStructures_CECT_segmentation/docs/README.md (hunk 7 above)
Comment (blacky-i) [reply models/renalStructures_CECT_segmentation/docs/README.md]: Done

File: models/renalStructures_CECT_segmentation/docs/README.md (hunk 6 above)
Comment (blacky-i) [reply models/renalStructures_CECT_segmentation/docs/README.md:60]: Done

File: models/renalStructures_CECT_segmentation/docs/README.md (hunk 5 above)
Comment (blacky-i) [reply models/renalStructures_CECT_segmentation/docs/README.md]: Done

File: models/renalStructures_CECT_segmentation/docs/README.md (hunk 4 above)
Comment (blacky-i) [reply models/renalStructures_CECT_segmentation/docs/README.md]: Done. Added cross reference to Data section
